src/hotspot/share/runtime/threadHeapSampler.cpp

```diff
@@ -71,9 +350,7 @@ double ThreadHeapSampler::fast_log2(const double& d) {
   assert(FastLogNumBits <= 20, "FastLogNumBits should be less than 20.");
   const uint32_t y = x_high >> (20 - FastLogNumBits) & FastLogMask;
   const int32_t exponent = ((x_high >> 20) & 0x7FF) - 1023;
-
-  assert(_log_table_initialized, "log table should be initialized");
-  return exponent + _log_table[y];
+  return exponent + log_table[y];
```

Comment:
The pre-generated values look good to me. Just a small nit. You can decide whether to take this or not:

C++ will generate an error if you have too many elements in `log_table[FastLogCount] = { ...`,  but won't complain if you have not enough elements. Any uninitialized elements will take up the default value of 0.

To prevent cut-and-paste errors in the future, I would suggest adding something like

```
   assert(FastLogCount > 0 && log_table[FastLogCount-1] != 0, "table should be full");
```